extension/opampextension/opamp_agent.go

```diff
@@ -0,0 +1,205 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opampextension
+
+import (
+	"context"
+	"os"
+	"runtime"
+
+	"github.com/oklog/ulid/v2"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opamp-go/client"
+	"github.com/open-telemetry/opamp-go/client/types"
+	"github.com/open-telemetry/opamp-go/protobufs"
+)
+
+// TODO: Replace with https://github.com/open-telemetry/opentelemetry-collector/issues/6596
+const localConfig = `
+exporters:
+  otlp:
+    endpoint: localhost:1111
+receivers:
+  otlp:
+    protocols:
+      grpc: {}
+      http: {}
+service:
+  pipelines:
+    traces:
+      receivers: [otlp]
+      processors: []
+      exporters: [otlp]
+`
+
+type opampAgent struct {
+	cfg    *Config
+	logger *zap.Logger
+
+	agentType    string
+	agentVersion string
```

Comment:
Should this be defined within the OpAmp package instead of by the extension?

Reply:
This will be the Collector's version, likely taken from [`component.BuildInfo`](https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/build_info.go#L16).

Reply:
Version differences between OpAMP implementations should be [handled by different capabilities](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#interoperability), which each have their own stability level. Currently there's no way to specify which version of the spec an implementation is using. The OpAMP spec should make [fairly strong guarantees](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#protobuf-schema-stability) about stability, but please feel free to say if you think something is missing. Protobuf itself should [gracefully handle](https://protobuf.dev/programming-guides/proto3/#updating) additive changes to the schema, so clients/servers with differing versions should be able to talk to each other.

Reply:
+1 to what @evan-bradley said. We do not expect to have OpAMP protocol version-number-specific code here. Any differences between client and server implementations should be handled via capability negotiation and Protobuf's additive change-ability.